pkg/envoy/cds/response.go

```diff
@@ -68,26 +68,17 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 		clusters = append(clusters, getTracingCluster(cfg))
 	}
 
-	resp := &xds_discovery.DiscoveryResponse{
-		TypeUrl: string(envoy.TypeCDS),
-	}
-
 	alreadyAdded := mapset.NewSet()
+	var resources []types.Resource
 	for _, cluster := range clusters {
 		if alreadyAdded.Contains(cluster.Name) {
 			log.Error().Msgf("Found duplicate clusters with name %s; Duplicate will not be sent to Envoy with XDS Certificate SerialNumber=%s on Pod with UID=%s",
 				cluster.Name, proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
 			continue
 		}
 		alreadyAdded.Add(cluster.Name)
-		marshalledClusters, err := ptypes.MarshalAny(cluster)
-		if err != nil {
-			log.Error().Err(err).Msgf("Failed to marshal cluster %s for Envoy with XDS Certificate SerialNumber=%s on Pod with UID=%s",
-				cluster.Name, proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
-			return nil, err
-		}
-		resp.Resources = append(resp.Resources, marshalledClusters)
+		resources = append(resources, cluster)
 	}
 
-	return resp, nil
+	return resources, nil
```

Comment:
This is also really tough to understand for a newcomer.
I see that the return type of the function is `[]type.Resource` and the function returns `resources`.
So this is great for the compiler, but I still don't know what this is.
I see that we are adding `cluster` to this - and from that I can start to understand that this is a collection of most likely CDS responses.
Is there a way to give more semantics to the slice name?
Because the function here is `cds.NewResponse()`, could this be `clusters` or `cdsResponses` or `xdsClusters`?
My intent is to make this easier to read - how do we stuff more context in this one line so one does not need to read the rest of the function to understand what is being returned.

Reply:
I can definitely try to add more context on the response variables if you think current code presents readability problems.